cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -361,16 +376,28 @@ writeStakeAddressInfo mOutFile delegsAndRewards =
       handleIOExceptT (ShelleyQueryCmdWriteFileError . FileIOError fpath)
         $ LBS.writeFile fpath (encodePretty delegsAndRewards)
 
-writeLedgerState :: ToJSON (NewEpochState ledgerera)
+writeLedgerState :: forall era ledgerera.
+                    ShelleyLedgerEra era ~ ledgerera
+                 => Ledger.UsesTxOut ledgerera
+                 => ToJSON (Ledger.NewEpochState ledgerera)
+                 => Ledger.ShelleyBased ledgerera
```

Comment:
Hmm. Why do we need these constraints here? Can't we write out a ledger state using the surface parts of the API rather than needing to go down a layer?

Perhaps we just need the CBOR or JSON instances for the `Query.LedgerState era`?

Generally we should be aiming to write most of the CLI without needing to look at internals from `Cardano.Api.Shelley`.